[CS2103T-T13-1] ORGANice

iskandarzulkarnaien opened the pull request:
@WilliamRyanK
@SimNee
@C-likethis123
@Yen-Peng

iskandarzulkarnaien committed:
Remove codacy badge

iskandarzulkarnaien committed:
Remove codacy badge

WilliamRyank committed:
Update Badges in README.adoc

WilliamRyank committed:
Merge pull request #7 from AY1920S1-CS2103T-T13-1/WilliamRyank-patch-1

Update Badges in README.adoc

iskandarzulkarnaien committed:
Update ContactUs Page (#27)

Yen-Peng committed:
Update AboutUs.adoc #1 (#11)

* Update About Us doc.

* Update About Us images.

* Update AboutUs.adoc

Amended responsibilities.

* Edit team description.

WilliamRyank committed:
Update Developer Guide (#24)

* Update Developer Guide

* Edit title

* Fix grammar

* Remove product survey

* Fix typo

* Fixing another grammar

C-likethis123 committed:
Update UserGuide.adoc #1 (#31)

* Update user guide contents to our application

* Add more content to user guide

Previously, the guide has incorrect descriptions of commands.
Lists were also not prefaced by sentences that introduce them.

This commit rectifies incorrect descriptions, based on feedback.
It also adds more sentences to improve readability.

* Update user guide to reflect changes to features

Changes to match, sort, process and done commands were made.

* Edit Command Summary of User Guide

* Edit typos in user guide

* Edit command instruction and application name

* Fix broken link for List of Attributes

WilliamRyank committed:
Change app name to ORGANice (#32)

* Travis testing

* Travis testing

* Update coveralls

* Edit coverage badge

* Change app name

SimNee committed:
Update ReadMe (#34)

* Travis test

* Travis test

* Test

* Formatting the ReadMe

* Formatting the ReadMe

* Updating the UI image

* Motivation

* Technology used

* Headings and UI

* comments and minor changes

* comments and minor changes

* Badges and UI changes

* Badges and UI changes

* remove extra space

* Add spaces after `==`

* Adding full stops

weikiat97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,33 +297,47 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to list patients/donors/doctors
```
Try to be consistent in adding fullstops or not! :)

Xelyion commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,33 +297,47 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |sort the potential donors by success rate/priority|match a patient with the best possible donor
+
+|`* * ` |user |add multiple data using external file|reduce the time to add in a data into the system
```
Formating error leads to asterisks displaying as '* * ' instead of * *

SebastianLie replied:
"Multiple Data" and "external file" are vague terms, perhaps consider specifying what kind of data and what format it should be in?

SebastianLie replied:
"Multiple Data" is a vague term, perhaps consider specifying what kind of data?

weikiat97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,33 +297,47 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+1.  User requests to list patients/donors/doctors
+2.  ORGANice shows a list of patients/donors/doctors
+3.  User requests to delete a specific patient/donor/doctor in the list
+4.  ORGANice deletes the patient/donor/doctor
```
Could include a confirmation feedback to the user when the commands are completed.

SebastianLie commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,33 +297,47 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |see list of all patients/donors/doctors|
+
+|`* * *` |careless user |edit a patient/donor/doctor|
```
I think that it isn't just careless users that want to edit information! Perhaps consider taking out the word careless

SebastianLie replied:
I think that it isn't just careless users that want to edit information! Perhaps consider taking out the word careless

weikiat97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,13 +348,31 @@ Use case ends.
+1.  User requests to list potential donors for a particular patient
+2.  ORGANice shows a list of potential donors for the patient
+3.  User requests to process the patient with a specified donor
+4.  ORGANice matches the patient and the specified donor
```
Should there be anything that happens before the use case ends?

Xelyion commented on docs/DeveloperGuide.adoc:
```diff
@@ -349,6 +381,9 @@ _{More to be added}_
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  The system should be usable by a novice.
```
How is novice defined? A computer novice or a novice at using the program.

weikiat97 reviewed (commented):
Can consider adding guarantee to your use cases.

WilliamRyank committed:
Remove unnecessary fields and add important fields for Person class (#75)

* Add William's refactor-person-class branch to Team Repo (#38)

* Remove email field

* fix checkstyle error

* Remove tags field

* Add nric field to person

* Add person type field

* Fixing some style

* Fix ordering

* Fix bug

iskandarzulkarnaien committed:
Fix typo in model address book (#78)

* Add William's refactor-person-class branch to Team Repo (#38)

* Remove email field

* fix checkstyle error

* Remove tags field

* Add nric field to person

* Add person type field

* Fixing some style

* Fix ordering

* Fix bug

* Fix typo in AddressBook.java

Yen-Peng committed:
Amend comments (#81)

SimNee committed:
Bloodtype field (#82)

* Setting up structure for bloodtype

* checkstyle errors, line too long

* added test files

* implement case insensitive

C-likethis123 committed:
Add Age Field for Patients (#79)

* Add William's refactor-person-class branch to Team Repo (#38)

* Remove email field

* fix checkstyle error

* Remove tags field

* Add nric field to person

* Add person type field

* Fixing some style

* Add Age Field for Patients

Previously there was:
* no differentiation between the types of Persons
* no differentiation between the types of PersonBuilder
* no change in how different types of Persons are displayed
* no difference in parsing

This commit:
* adds a Patient class which is a Person but with additional criteria
* adds a PatientBuilder class
* different UI displays based on type of Person
* parses Patients and normal Persons differently

* Improve coding style of AddCommandParser.java

* Rectify problems pointed out in PR Review

The following problems are rectified:
* Changed description to 'a valid age'
* Changed condition for a valid age
* Changed MESSAGE_CONSTRAINTS to reflect new conditions

* Change unit test for Age.java in AgeTest.java

* Edit PatientTest.java according to PR Review

* Fix failing build by removing unused import

* Edit constant names to improve coding style

* Introduce constants to improve readability

* Remove HEAD arrows

C-likethis123 committed:
Fix typo in user guide (#87)

Remove extra NRIC field in command format for doctors

WilliamRyank committed:
Add donor class (#84)

* Add donor class

* Change test cases

* Add equals and hash function for donor class

* Checkstyle fix

Yen-Peng committed:
Add organ class (#86)

* Add organ class and organtest class.

* Amend comments.

* Add more test cases.

* Fix checkstyle errors.

* Fix failed test.

* Remove unnecessary test case.

* Remove unused import.

WilliamRyank committed:
Add doctor in charge field and improve test case coverage (#85)

* Add doctor in charge field

* Fix extra line

* Improve test code coverage

* Add Set of Bloodtypes in BloodType class

* Add test cases for DonorTest

SimNee committed:
Bloodtype field (#89)

* Setting up structure for bloodtype

* checkstyle errors, line too long

* added test files

* implement case insensitive

* more test cases added

WilliamRyank committed:
Add expiry date field (#98)

* Add expiry date class

* Add OrganExpiryDate class

* Add test case for Organ Expiry Date

* Remove a date check

* Change date parse implementation

* Add more testcases

iskandarzulkarnaien committed:
Add doctor class (#97)

* Update AddCommandParser.java with doctor class support

* Fix typo in AddCommandParser

* Add Doctor class

* Update JsonAdaptedPerson

* Add DoctorCard class

* Update PersonListPanel

* Change ordering of person type checks to alphabetical order

* Change ordering of checks AddCommandParser to alphabetical

* Change ordering of checks in PersonListPanel to alphabetical

* Add DoctorListCard.fxml

* Fix typo in Type.java

* Add DoctorBuilder for tests

* Update TypicalPersons to reflect Doctor class

* Add DoctorTest

* Add toModelType test for Doctor class

* Update AddCommandTest for Doctor class

* Fix misc checkstyle issues

* Fix AddCommandTest doctor variable

* Fix Checkstyle Issues

* Change Person type references to Doctor in DoctorTest

* Reverse order of equality check in CommandTestUtil (see issues #99)

iskandarzulkarnaien committed:
Refactor packaging to organice (#100)

iskandarzulkarnaien committed:
Change Typical Persons naming format to `PERSONTYPE_PERSONNAME` (#101)

* Change Typical Persons naming format to `PERSONTYPE_PERSONNAME`

* Fix checkstyle issues

* Fix missing whitespace in Doctor and Patient Test classes

* Correctness fix for `PERSON_AMY` to `DOCTOR_AMY` and `PERSON_BOB` to `PATIENT_BOB`

* Fix checkstyle

C-likethis123 committed:
Create Priority field (#88)

* Create Priority field

* Integrate Priority field to JsonAdaptedPerson

* Add dummy priority in AddCommandParser.java

* Make changes to test cases

* Change UI to display priority

* Add equality check for Patient object (fixes #90) 

* Fix failing build

* Add age in equality check

* Make requested changes before PR

* Fix error in JsonAdaptedPerson

Previously the JsonAdaptedPerson constructor's method signature
throws an IllegalValueException, causing JsonSerializableAddressBook
to not work.

This commit removes the error thrown because the constructor is
not supposed to throw the error

* Utilise Doctor and DoctorBuilder in tests

* Improve coding style to fix build

SimNee committed:
Tissue type field (#96)

* tissue type field

* test for duplicates and invalide tissuetypes more than 12

* Change package name

* Fix camel case

WilliamRyank committed:
Integrate fields to patient and donor (#107)

* tissue type field

* test for duplicates and invalide tissuetypes more than 12

* Change package name

* Fix camel case

* Integrate donor and patient

* Change implementation of containsDoctor

Yen-Peng committed:
Add list feature (#113)

* Edit message in List Command class.

* Amend checkstyle errors.

* Edit execute method in List Command class.

* Edit ListCommand & parser class based on pr feedback.

* Add test cases

* Remove try-catch block in ListCommand class.

* Creating constants and fix logic

* Amend isTypePresent method in ListCommandParser class.

* Remove unused import.

* Modify parseType method in ListCommandParser class.

* Remove unused imports.

WilliamRyank committed:
Update user guide for match and list commands (#117)

* Update user guide

* Fix typo

* Fix typo

WilliamRyank committed:
Update ug (#126)

* Add UG for add feature

* Add stuff

* Ammend UG

* Fix UG

* Fix sentences

* Change Picture

WilliamRyank committed:
DG update (#129)

* DG update

* DG for add form mode

* Fix typo

* Update DeveloperGuide.adoc

Yen-Peng committed:
Update ugdg (#131)

* Update list and sort features ugdg.

* Update ModelClassDiagram.puml

* Add SortSequenceDiagram.puml

C-likethis123 committed:
Add Match Feature to Developer and User Guide (#130)

* Add Match Implementation to Developer Guide

* Edit user guide

SimNee committed:
Ready to merge processing user guide info (#128)

* processing user guide info

* Adding diagrams for sort

Yen-Peng committed:
Update logo and app name in GUI title box. (#136)

SimNee committed:
Ugdg read me (#135)

* processing user guide info

* Adding diagrams for sort

* changes to the sort in DG

* Processing DG

C-likethis123 committed:
Edit user guide (#137)

* Fix typo in user guide

* Edit Introduction and Features Sections

C-likethis123 committed:
Implement minimal version of Matching (#115)

* Create MatchCommandParser to parse MatchCommands

MatchCommands are meant to match patients to suitable donors
in the system.

* Add function to parse match commands

* Add skeleton of MatchCommand

* Add matching functionality to MatchCommand

This is an intermediate implementation of MatchCommand, which
is executed to match a patient with Donors in the Donor pool.
Currently when this command is executed, it displays all donors which
matches the specified patient.

Currently in this implementation, a match happens when patient and donor's
blood types are compatible and they pass the tissue typing test.

* Modify AddressBook.java to find and return Patient

* Retrieves specified Patient from PersonList

* Add blood type tests

* Add tissue typing tests

* Fix broken build

* Make edits based on PR Review

* Add unit test for MatchCommand

* Create a MatchCard to display Match Results

* Update PersonList with Match Results

* Render MatchCard to display matches

* Store results of matches

* Add NumberOfMatches to track matches of Patients

* Edit PersonListPanel to display matches

* Add MatchedDonor to represent matching donors

* Add MatchedPatient to represent number of matches

A MatchedPatient contains a Patient's details and the number
of matching donors with the Patient

* Edit Donor and Patient match result details

Previously, match results are stored in the Donor and Patient class.
With this commit, match results are no longer stored in
Donor and Patient classes, but are stored in MatchedDonor
and MatchedPatient classes.

* Change PersonListPanel for MatchCommands

Previously, matches and people were stored in the Logic's
filteredPersonsList. This resulted in bugs when rendering the
matches and people.

The solution is to create a separate list to store matches. The
MainWindow will select the list to display.

* Add a new list to store matches

* Add Matching functionality

The changes support matching individual Patients and all Patients

* Fixed failing build

* Add unit tests for MatchCommandParser.java

* Fix coding style

* Make change to messages

* Add additional tests to ModelManagerTest.java

* Add unit tests for MatchedPatient.java

* Make changes to MatchedPatient equals method

* Add unit test for MatchedDonor.java

* Edit equality check for MatchDonor

* Add unit tests for AddressBook and UniquePersonList

* Add tests for LogicManager

* Add integration tests for CommandResult and MatchCommand

* Modify test data for correct matching test results

* Modify ModelManager methods and CommandTestUtil

C-likethis123 committed:
Edit user guide (#138)

* Fix typo in user guide

* Edit Introduction and Features Sections

* Change tone of user guide, include labelled diagrams

* Make changes according to reviews

* Make typo changes

SimNee committed:
Status field (#139)

* Status field

* why got errors in test case

* i hope the test case and the codestyle is fixed

* Checkstyle please work

* test cases issues

* removed parsestatus and minor edits

* unused imports

* Status.STATUS_NOT_PROCESSING changed

Yen-Peng committed:
Add diagrams & update ugdg (#133)

* Update list and sort features ugdg.

* Update ModelClassDiagram.puml

* Add SortSequenceDiagram.puml

* Fix typo in model class diagram

* Add list activity diagram.

* Fix checkstyle.

* Amend diagram.

* Fix title markdown.

* Amend dg.

* Add spacing in dg.

* Fix diagrams spacing.

* Update link to user guide in help window.

C-likethis123 committed:
Remove redundant persons from TypicalPerson (#149)

* Remove Benson from TypicalPersons

* Remove Persons from TypicalPersons

* Change test cases to remove redundant persons

WilliamRyank committed:
Update UG (#148)

* DG update

* UG for add command

* Fix typo

* fix typo

* Change language

* Fix

* Fix UG

* Fix

* Fix

* Fix typo

WilliamRyank committed:
Add command form (#134)

* Adding form type

* Minimal doctor form

* Fix bug

* Add exception for invalid input

* Fixing doctor form fxml

* Manage Form Logic

* Add Patient form

* Add patient form and logic

* Fix test cases

* Finish donor form

* Checkstyle fix

* Add progress bar animation

* Add find match button for patient

* Fixing ui bugs

* Add undo feature

* Fix bug

* Add log

* checkstyle fix

* Fix

C-likethis123 committed:
Improve wording of Developer and User Guide  (#150)

* Add Match Implementation to Developer Guide

* Change wording of User Guide

* Add Preambles to Developer Guide

Yen-Peng committed:
Add sort feature (#145)

* Add sort feature.

* Add comparators.

* Add sort command and parser.

* Update sort list display on window.

* Fix checkstyle.

* Fix checkstyle.

* Fix test checkstyle.

* Fix checkstyle.

C-likethis123 committed:
Change dummy data to be populated (#157)

Dummy data created when a person first uses ORGANice is changed.

C-likethis123 committed:
Modify existing AB3 diagrams  (#152)

* Edit diagrams given in existing AB3 Guide

* Improve explanation of match feature

Changes to the explanation are as follows:
* Discuss different aspects in the Logic, UI and Model section.
* Include code snippets for better understanding

To Do:
* Add sequence diagrams
* Add class diagrams
* Expand on design considerations

* Edit developer guide

C-likethis123 committed:
Enhancements to match feature (#141)

* Add more messages detailing result of each match

Previously there were only messages detailing if
the command succeeded or failed.

Now some new types of messages are added:
- If the command matches a specific patient, the message
details which patient was matched with all donors.
- If no matches were found, the resulting message states that.
- Messages to provide details on how many matches, and for which patient

* Beautify UI of displayed matches

The following changes were made to the UI:
* Success rate is to be displayed in 2 decimal places instead of one long string
* Match results to be displayed in red
* Nric of patient and donor to be prefixed with "Nric: " followed
by the NRIC of the person.

* Create new ModelStub for testing

Previously a private ModelStub class was created for both AddCommandTest
and MatchCommandTest. This resulted in duplication of code whenever the
Model implementation changed.

This ModelStub class will now be used by all classes, removing the duplication
of code.

* Edit messages according to PR review

* Edit tests based on changed messages

* Edit test cases based on changes

* Change error thrown in getMatchList

* Changed AssertionError to assert statements

SimNee committed:
WIP: Crud - update (#119)

* WIP simple update

* WIP minimal edit from index to nric done

* Simple edit with messages amendments

* WIP fixing test cases

* Removed some irrelevant test cases and changed edit command to not allow the edit of NRIC and type

* Fix command usage

* Fix checkstyle and test

* Fix error

* Fix test cases

* Remove unused import

* Change

* Fix checkstyle

iskandarzulkarnaien committed:
Implement basic find by prefix feature (#143)

* WIP: change findcommandparser

* Implement find by prefix

* Fix ModelManagerTest and fix error when user key in multi-word prefix inputs

* Fix tests

* Fix checkstyle

* Refactor PersonContainsPrefixesPredicate#test by extract method

* Fix displaying people who do not possess input attributes at all

* Fix typo in PersonContainsPrefixesPredicate

* Add guard clause to detect wrong find command preamble

* Fix tests

* Fix checkstyle

* Change tissuetype finding to be irrespective of order

* Add null check for FindCommand predicate and remove errant print statements

* Fix empty tissue types being parsed into list containing empty string

* Change string literal "find" to FindCommand.COMMAND_WORD

WilliamRyank committed:
Update class diagram (#160)

* DG update

* Change class diagram

iskandarzulkarnaien committed:
Update User Guide for Find by Prefix Feature (#158)

* Add screenshot to showcase find by prefix feature

* Update find command section in User Guide

* Update find command section in User Guide

C-likethis123 committed:
More edits to developer guide (#159)

* Edit diagrams given in existing AB3 Guide

* Improve explanation of match feature

Changes to the explanation are as follows:
* Discuss different aspects in the Logic, UI and Model section.
* Include code snippets for better understanding

To Do:
* Add sequence diagrams
* Add class diagrams
* Expand on design considerations

* Edit developer guide

* Add on to developer guide

* Make changes according to PR

Yen-Peng committed:
Update ugdg (#153)

* Update model class diagram image.

* Remove API links in dg.

* Update ug.

* Update sort sequence diagram.

* Fix checkstyle.

* Amend english.

* Change bullets.

iskandarzulkarnaien committed:
Update developer guide with rough workflow of Find (#162)

* Update DG with rough workflow of Find

* Update DG with rough workflow of Find

SimNee committed:
Implement minimal version of Processing Command (#127)

* WIP: ProcessingCommand

* Adding method to check if patient and donor exist

* Able to process 2 ic/ and return the default list out

* delete unused imports

* fixed issues regarding processing and matching

* null pointer errors arise

* remove the using of storage for the tasklist

* not able to update list

* markAsDone is done

* made setter method for processinglist

* setter changes

* processingmark command done

* some test cases added

* checkstyle

* checkstyles

* Model issues

* Model issues

* null pointer fixes

* null pointer fixes

* checkstyle

* checkstyle

* checkstyle

* command completed. Test cases not yet

* trying to revert the conflict removal

* command completed. Test cases not yet

* command completed. Test cases not yet

* command completed. Test cases not yet

SimNee committed:
Done-command (#155)

* test cases not ready

* added parser

* eof added

* checkstyle issues half fixed

* checkstyles

* ammended from comments

* eof added

* trying to revert the conflict removal

* trying to revert the conflict removal

* trying to revert the conflict removal

* trying to revert the conflict removal

* trying to revert the conflict removal

* command completed. Test cases not yet

C-likethis123 committed:
Remove mention of positive blood types (#168)

* Remove mention of positive blood types

* Edit test cases to remove positive signs

iskandarzulkarnaien committed:
Improve dummy data generation (#172)

* Add dummy data generation in SampleDataUtil

* Fix checkstyle

WilliamRyank committed:
Remove delete (#171)

* DG update

* Remove command

* Remove delete command

* fix build

* fix typo

C-likethis123 committed:
Minor UI edits (#169)

* Increase size of Message Display Box

* Modify ListCards

Changes made:
* NRIC is displayed as a normal attribute, not a tag
* Status and Priority is displayed as a tag

* Change colour according to value

Priority colour scheme is as follows:
* Red for High Priority
* Orange for Medium Priority
* Green for Low Priority

Status colour scheme is as follows:
* Dark blue for not processing
* Light blue for processing

WilliamRyank committed:
Add valid criteria prompt (#173)

* DG update

* Add message

* Resize

* Fix build

Yen-Peng committed:
Enhance sort and add test cases (#170)

* Correct success rate comparison.

* Add match and name comparators.

* Update testcases halfway.

* Remove test cases to merge the rest first.

* Fix checkstyle.

* Update ug.

* Update ug.

* Update dg for sort.

* Update sort dg.

* Update dg.

* Fix exception issue.

* Update list dg.

* Fix checkstyle.

C-likethis123 committed:
Edit user guide (#178)

* Fix typo in user guide

* Edit Introduction and Features Sections

* Change tone of user guide, include labelled diagrams

* Make changes according to reviews

* Make typo changes

* Change exit program section

iskandarzulkarnaien committed:
Change default status to NOT_PROCESSING and fix age exceeding limit (#180)

iskandarzulkarnaien committed:
Improve basic find (#184)

* Update contains phrase method to check for exact phrase

* Update user guide for find to reflect changes

iskandarzulkarnaien committed:
Change OrganExpiryDate default data generation to be random (#182)

iskandarzulkarnaien committed:
Consolidate all display changes to displayedPersonList (#181)

* Consolidate all display changes to displayedPersonList

* Fix checkstyle

* Fix tests and checkstyle

WilliamRyank committed:
Fix form bug (#185)

* DG update

* Fix bug

C-likethis123 committed:
Fix typo error in 'Priority' error message (#234)

Added a space. This commit fixes #220.

C-likethis123 committed:
Change range of valid ages (#233)

* New range of valid ages is: 0 (inclusive) to 150 (inclusive)

* Message Usage elaborates on valid age inputs

C-likethis123 committed:
Add signed bloodtypes (#231)

* Add support for signed bloodtypes (eg A+, B-).

* Fix test cases for tests involving bloodtype values

The change is needed because of change in blood types. Now blood
types are accompanied with the signs '+' or '-'.

* Rename #isBloodTypeMatch to #isCompatibleBloodType

Yen-Peng committed:
Fix list issue #187 (#236)

* Fix list issue #187.

* Fix checkstyle.

* Fix checkstyle.

* Update test cases.

* Fix checkstyle.

* Fix list test cases.

* Amend list command class.

* Amend list command class.

* Fix checkstyle.

C-likethis123 committed:
Fix matching related bug (#232)

* Add validation to user input on matches

Previously, when more than one value has the same prefix,
the app takes in the last value only. This is a feature flaw
which is fixed in this commit.

This commit rejects user inputs if there is more than one value
with the 'ic/' prefix.

* Add test cases that rejects multiple inputs of same prefixes

* Add more explanation on MatchCommand's message usage

SimNee committed:
Processing command (#176)

* WIP: ProcessingCommand

* Adding method to check if patient and donor exist

* Able to process 2 ic/ and return the default list out

* delete unused imports

* fixed issues regarding processing and matching

* null pointer errors arise

* remove the using of storage for the tasklist

* not able to update list

* markAsDone is done

* made setter method for processinglist

* setter changes

* processingmark command done

* some test cases added

* checkstyle

* checkstyles

* Model issues

* Model issues

* null pointer fixes

* null pointer fixes

* checkstyle

* checkstyle

* checkstyle

* command completed. Test cases not yet

* trying to revert the conflict removal

* command completed. Test cases not yet

* command completed. Test cases not yet

* command completed. Test cases not yet

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* fix issue #106 regarding addCommand tissuetyping

* fix issue #106 regarding addCommand tissuetyping

* fix issue #206 regarding addCommand tissuetyping

* added more exception handlings

* merge conflict

* merge conflict

* added more exceptions handling and fixed #207

* added more exceptions handling and fixed #207

* more exception handlings

* more exception handlings

* more exception handlings

iskandarzulkarnaien committed:
Implement Fuzzy Search (#174)

* WIP: Implemented calculation of Levenshtein distance

* Rename PersonContainsExactPrefixesPredicate to PersonContainsPrefixesPredicate

* Fix LD calculation always 0

* Add method to find min LD of name

* WIP Name split find LD

* Update fuzzy to process multiword names

* WIP: Send sorted arraylist to model

* Working build of Fuzzysearch

* Cleanup of FuzzyFindCommand

* Fix tests and checkstyle

* Change `find` to use fuzzy search and normal search to be `exactfind`

* Shift `#calculateLevenshteinDistance` to `StringUtil`

* Update UG and fix checkstyle

* Update findByPrefix.png

* [Fuzzy Search] Rename `find` to `exactFind` and `fuzzyFind` to `find`

* [Fuzzy Search] Fix checkstyle

* [Fuzzy Search] Change find to always be based on full person list

SimNee committed:
Done command (#230)

* added parser

* checkstyles

* amended javadoc comments

* trying to revert the conflict removal

* command completed. 

* adding the result of done command, pass or fail

* checkstyle and javadoc

* adding more exceptions

iskandarzulkarnaien committed:
Fix date generation (#251)

* [Fix Random Date Generation] Add proper brackets in int typecast expression

* [Fix Random Date Generatio] Actually use the random date

* Change date range increment to [30, 150)

* Increase readability of date generation

* [Date Gen Fix] Fix checkstyle

WilliamRyank committed:
Give more error message to doctor ic not found (#249)

* DG update

* fix

* Fix build:

C-likethis123 committed:
Edit error message thrown when attempting to add duplicate (#242)

* Modify error message for clarity

Previously, there was an error message shown when a user
attempts to add a duplicate.
This commit elaborates the error message to prompt the user
to change the NRIC input.

* Change user guide
The following changes to the user guide were made:
    -Improve documentation of add feature
    -Delete 'Detecting Duplicates' section
    -Add clarification for tissue types

iskandarzulkarnaien committed:
Type check find (#253)

* [Type Check Find] WIP add type check util class

* [Type Check Find] WIP check util

* [Type Check Find] Add type check, remove fuzzy for type

* [Type Check Find] Fix checkstyle

* [Type Check Find] If no fuzzy params, dont match. Check prefixes in parser

* [Type Check Find] Remove unused import

* [Type Check Find] Fix error msg

* [Type Check Find] Fix FindCOmmandParser

* [Type Check Find] Correct preamble length check

* [Type Check Find] Fix tests and checkstyle

SimNee committed:
Done command (#261)

* test cases not ready

* added parser

* eof added

* checkstyle issues half fixed

* checkstyles

* ammended from comments

* eof added

* trying to revert the conflict removal

* trying to revert the conflict removal

* trying to revert the conflict removal

* trying to revert the conflict removal

* trying to revert the conflict removal

* command completed. Test cases not yet

* adding the result of done command, pass or fail

* adding the result of done command, pass or fail

* checkstyle and javadoc

* removing import deleteCommand in addressbookParser

* checkstyle

* adding more exceptions

* amendments to isPass method

* amendments to isPass method

* amendments to matching command for a fail cross-matching

* amendments to matching command for a fail cross-matching

* exception throwing unknowingly why

* fixed the issue regarding the result of done command not saving in match

* fixed the issue regarding the result of done command not saving in match

* fixed the issue regarding the result of done command not saving in match

* fixed the issue regarding the result of done command not saving in match test cases

* fixed the issue regarding the result of done command not saving in match test cases

* changes from reviews

* pr reviews fixes

* pr reviews fixes

* pr reviews fixes

* markasdone bug

iskandarzulkarnaien committed:
Validate Nric Checksums (#255)

* [Add NRIC Checksum Validation] Add checksum validation

* [Add NRIC Checksum Validation] Fix sampledata generation

* [Validate NRIC] Fix tests and checkstyle

C-likethis123 committed:
Improve documentation for match command (#241)

The following sections were changed:
* Introduction 
* Match Feature
* Change tone of user guide
* Exit

Other changes include:
* Adding labelled diagrams

Yen-Peng committed:
Update list & edit ug (#244)

* Update ug for list and edit features.

* Amend changes.

* Update dg.

* Amend grammar in ug.

WilliamRyank committed:
Bug ui form (#247)

* DG update

* Fix ui bug

* Fix

* Fix

* Fixbuild

* Fix ui bugs

* UG

* UG

* Fix doctor ic validation

* Fix some error

* Change screenshot

WilliamRyank committed:
Add delete command (#260)

* DG update

* Delete command

* Add error message

* Fix

C-likethis123 committed:
Change explanation of match feature (#258)

* Edit diagrams given in existing AB3 Guide

To Do:
* Add sequence diagrams
* Add class diagrams
* Expand on design considerations

* Rewrite implementation details of MatchCommand

* Add match feature example usage scenario

SimNee committed:
adding other attributes to edit (#177)

* adding other attributes to edit

* getter and setter methods of the attributes for

* getting values of attributes in editCommandParser

* trying to edit

* removing organ exp date for editing

* Fix edit

* Add error message

SimNee committed:
Processing command (#267)

* WIP: ProcessingCommand

* Adding method to check if patient and donor exist

* Able to process 2 ic/ and return the default list out

* delete unused imports

* fixed issues regarding processing and matching

* null pointer errors arise

* remove the using of storage for the tasklist

* not able to update list

* markAsDone is done

* made setter method for processinglist

* setter changes

* processingmark command done

* some test cases added

* checkstyle

* checkstyles

* Model issues

* Model issues

* null pointer fixes

* null pointer fixes

* checkstyle

* checkstyle

* checkstyle

* command completed. Test cases not yet

* trying to revert the conflict removal

* command completed. Test cases not yet

* command completed. Test cases not yet

* command completed. Test cases not yet

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* added more checks to ensure valid patient and donor pair

* fix issue #106 regarding addCommand tissuetyping

* fix issue #106 regarding addCommand tissuetyping

* fix issue #206 regarding addCommand tissuetyping

* added more exception handlings

* merge conflict

* merge conflict

* added more exceptions handling and fixed #207

* added more exceptions handling and fixed #207

* more exception handlings

* more exception handlings

* more exception handlings

* bugs

* checkstyle

C-likethis123 committed:
Validate type adding (#263)

Refactor AddCommandParser

* Code for checking if form mode should be launched is
consolidated in the #isLaunchForm method.

* Validation for one value per prefix is done in the

* Add test cases to test for type validation

C-likethis123 committed:
Enhance match feature (#264)

* Rename class names for clarity

Previously, MatchedDonor objects are displayed by the UI component
MatchCard. This is not an intuitive name.

Likewise, NumberOfDonors card is renamed to MatchedPatientCard.

* Change success rate to compatibility rate

* Fix test cases

C-likethis123 committed:
Add object diagrams to explain match feature (#268)

* Rewrite implementation details of MatchCommand

* Add match feature example usage scenario

* Add Object Diagrams for usage scenario

SimNee committed:
UG changes for processing (#269)

* UG changes for processing

* grammar errors

C-likethis123 committed:
Update 'Proposed new extensions' (#270)

* Edit Introduction and Features Sections

* Change tone of user guide, include labelled diagrams

* Change exit program section

* Improve explanation for match feature

* Update 'Proposed New Features'

Change all references to success rate to 'compatibility rate'

C-likethis123 committed:
Change screenshots generated for match command (#273)

Change screenshots generated for Match feature

iskandarzulkarnaien committed:
Add tests find (#274)

* [Add Tests Find] Add tests for calculateLevenshteinDistance

* [Add Tests Find] Add tests for FindCommandParser

* [Add Tests Find] Simplify assertions in ExactFindCommandTest

* [Add Tests Find] Add tests for FindCommand

* [Add Tests Find] FixCheckstyle

WilliamRyank committed:
Add UG for clear and delete command (#266)

* DG update

* UG

* Fix

* Fix

* Fix

Yen-Peng committed:
Edit my ppp (#265)

* Add portfolio.

* Add tags to ppp.

* Remove extra project.

* Update help ug

* Add help tag to ppp.

* Update diagram caption.

* Add logging to sort feature.

* Update ppp.

* Update sort dg to remove setSort method.

* Amend list error message.

* Amend list diagram.

* Amend sort dg to change success rate to compatibility rate.

* Update sort ug success rate to compatibility.

* Update ppp.

* Change help ug.

* Add eof to ppp.

* Fix checkstyle.

* Update model stub.

* Fix compile test error.

* Amend ug to add mention of buttons.

* Update ppp.

* Update intro to add role.

* Update diagram name in ppp.

* Format ppp.

* Fix grammar.

* Check grammar.

* Make titles consistent.

* Add numbering.

* Update ppp.

* Add contribution.

WilliamRyank committed:
Change UI (#276)

* DG update

* Fic

SimNee committed:
Simnee ppp (#271)

* UG changes for processing

* grammar errors

* UI picture for readMe

* DG

* ppp

SimNee committed:
Simnee ppp (#278)

C-likethis123 committed:
Add logging to match feature (#259)

* Add logging to match feature

* Fix checkstyle and test cases

* Add logging statements to match methods

iskandarzulkarnaien committed:
Update dg find (#277)

* [Find DG] WIP, add some diagrams

* [Update Find DG and Isk PPP] wip PPP

* [Update Find DG and Isk PPP] WIp add diagrams and update DG

* [Update Find DG and Isk PPP] Final dg?

* [Update Find DG and Isk PPP] Update about us

* [Update Find DG and Isk PPP] PPP

* [Update Find DG and Isk PPP] add link to ab3

* [Update Find DG and Isk PPP] update links PPp

* [Update Find DG and Isk PPP] update forum posts links

* [Update Find DG and Isk PPP] update reviewed PR

* [Update Find DG and Isk PPP] change typos

* [Update Find DG and Isk PPP] add newline at eof

* [Update Find DG and Isk PPP] fix dg ppp

* [Update Find DG and Isk PPP] lowercase ppp

* [Update Find DG and Isk PPP] PPP reword

* [Update Find DG and Isk PPP] remove

* [Update Find DG and Isk PPP] remove blue

* [Update Find DG and Isk PPP] Add better paragraphing to ug

* [Update Find DG and Isk PPP] Replace Organice with ORGA__Nice__

* [Update Find DG and Isk PPP] Fix introduction page

* [Update Find DG and Isk PPP] Fix introduction page

* [Update Find DG and Isk PPP] Fix printing

WilliamRyank committed:
Update PPP (#154)

* DG update

* Add dg for add form

* Change size

* Add diagram

* Fix

* Add diagram

* Fix

* Fix

* Fix

* Test

* Fix

* Fix

* Fix

* Fix

* Fix

* A

* A

* Fix:

* Fix

* Fix

* Fix

iskandarzulkarnaien committed:
[Add Logging to Find] add logging (#279)

C-likethis123 committed:
Add integration tests (#275)

* Add more integration tests for MatchCommand

* Add integration tests for ProcessingCommand

* Add test cases for SortCommand

iskandarzulkarnaien committed:
[Fix Ui] fix (#280)

C-likethis123 committed:
Start drafting the PPP (#163)

* Start drafting PPP

* Make more edits to PPP

* Change 'Contribution' section

* Elaborated on enhancements I made

* Add more contributions

* Include links for PPP

* Edited wording of PPP

* Standardise Developer Guide

WilliamRyank committed:
Ppp (#282)

* DG update

* Fix
